packages/fxa-shared/db/models/auth/index.ts

```diff
@@ -80,24 +81,29 @@ export async function getAccountCustomerByUid(uid: string) {
 
 /**
  * Fetch the FxA user id and primary email for a customer by Stripe customer id.
- * @param customerId
- * @returns {Object} result
- * @returns {string} result.uid - The FxA user id
- * @returns {string} result.email - The primary email for the FxA user
  */
 export async function getUidAndEmailByStripeCustomerId(
-  customerId: string
-): Promise<{ uid: string; email: string }> {
+  customer: string | Stripe.Customer | Stripe.DeletedCustomer | null
+): Promise<{ uid: string | null; email: string | null }> {
+  if (typeof customer !== 'string') {
```

Comment:
Can we not accept non-strings in the parameter?  Might be confusing to people using this function when they pass in one of the accepted types only to get an error.

Reply:
That's what I originally had, but the consumer (now just `stripeUpdate` in `lib/payments/capability.ts`) passes in a Stripe `subscription.customer` which can be a Stripe `Customer` or `DeletedCustomer` as well. I was trying to avoid checking the type of the value passed into the helper in every caller (though there's just one caller currently).

Looking at this one caller, `stripeUpdate`, since this only comes from Stripe webhooks where `event.data.object` is a Stripe `subscription` with a non-expanded customer, I guess I could do something like:
```
({ uid, email } = await getUidAndEmailByStripeCustomerId(sub.customer as string));
```
What do you think @chenba ?